src/components/AnonymousChat/BaseChatItem.tsx

```diff
@@ -26,34 +27,46 @@ const styles = StyleSheet.create({
 
 const BaseChatItem = ({item, index, type}: BaseChatItemComponentProps) => {
   const {selectedChannel, handleUserName} = useChatScreenHook(type || ANONYMOUS);
-  const {anon_user_info_emoji_code, anon_user_info_color_code} =
-    selectedChannel?.rawJson?.channel || {};
-  const renderAnonymousIcon = (anonColor: string, anonEmoji: string) => (
+
+  const renderAnonymousAvatar = (anonColor: string, anonEmoji: string) => (
     <AnonymousIcon color={anonColor} emojiCode={anonEmoji} size={dimen.normalizeDimen(24)} />
   );
 
-  const renderDefaultIcon = () => (
-    <FastImage style={styles.containerPicture} source={{uri: item?.user?.profilePicture}} />
+  const renderSignedAvatar = () => (
+    <FastImage
+      style={styles.containerPicture}
+      source={{uri: item?.user?.profilePicture ?? DEFAULT_PROFILE_PIC_PATH}}
+    />
   );
 
-  const handleImageUser = () => {
+  const handleAvatar = () => {
+    const selectedJson = selectedChannel?.rawJson;
+    const itemJson = item?.rawJson;
+
     if (type === ANONYMOUS) {
-      if (item?.user?.username !== ANONYMOUS_USER) {
-        return renderDefaultIcon();
-      }
-      return renderAnonymousIcon(anon_user_info_color_code, anon_user_info_emoji_code);
+      if (!item?.isMe && item?.user?.username !== ANONYMOUS_USER) return renderSignedAvatar();
+
+      return renderAnonymousAvatar(
+        selectedJson?.channel?.anon_user_info_color_code,
+        selectedJson?.channel?.anon_user_info_emoji_code
+      );
     }
 
-    if (item?.rawJson?.anon_user_info_emoji_code) {
-      return renderAnonymousIcon(anon_user_info_color_code, anon_user_info_emoji_code);
+    if (itemJson?.anon_user_info_emoji_code) {
+      return renderAnonymousAvatar(
+        itemJson?.anon_user_info_color_code,
+        itemJson?.anon_user_info_emoji_code
+      );
     }
-    if (item?.rawJson?.user?.name === ANONYMOUS_USER) {
-      return renderAnonymousIcon(
-        selectedChannel?.rawJson?.anon_user_info_color_code,
-        selectedChannel?.rawJson?.anon_user_info_emoji_code
+
+    if (itemJson?.user?.name === ANONYMOUS_USER) {
+      return renderAnonymousAvatar(
+        selectedJson?.anon_user_info_color_code ?? selectedJson?.channel?.anon_user_info_color_code,
+        selectedJson?.anon_user_info_emoji_code ?? selectedJson?.channel?.anon_user_info_emoji_code
       );
     }
-    return renderDefaultIcon();
+
+    return renderSignedAvatar();
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The refactoring of the avatar rendering logic in `BaseChatItem.tsx` is well done. The new functions `renderAnonymousAvatar`, `renderSignedAvatar`, and `handleAvatar` are more readable and maintainable than the previous implementation. However, there seems to be a potential issue with the handling of anonymous avatars. In the case where `itemJson?.user?.name === ANONYMOUS_USER`, it appears that you're trying to use `selectedJson?.anon_user_info_color_code` and `selectedJson?.anon_user_info_emoji_code` as fallbacks if `selectedJson?.channel?.anon_user_info_color_code` and `selectedJson?.channel?.anon_user_info_emoji_code` are undefined. But if `selectedJson` itself is undefined, this will throw an error. You should add checks for `selectedJson` before accessing its properties.

```diff
- selectedJson?.anon_user_info_color_code ?? selectedJson?.channel?.anon_user_info_color_code,
- selectedJson?.anon_user_info_emoji_code ?? selectedJson?.channel?.anon_user_info_emoji_code
+ selectedJson ? (selectedJson.anon_user_info_color_code ?? selectedJson.channel?.anon_user_info_color_code) : undefined,
+ selectedJson ? (selectedJson.anon_user_info_emoji_code ?? selectedJson.channel?.anon_user_info_emoji_code) : undefined
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->